### ISSUE TITLE
Fix multi ENOSPC error

### ISSUE THREAD

**Issue by stetsmando** [stetsmando/pi-motion-detection#1]
When motion is detected the module fails and spits out:
```
motion!
mmal: mmal_vc_component_enable: failed to enable component: ENOSPC
mmal: camera component couldn't be enabled
mmal: main: Failed to create camera component
mmal: Failed to run camera app. Please check for firmware updates


motion!
mmal: mmal_vc_component_enable: failed to enable component: ENOSPC
mmal: camera component couldn't be enabled
mmal: main: Failed to create camera component
mmal: Failed to run camera app. Please check for firmware updates
```

Not sure what's going on, but I think it's related to the various child processes and how they communicate with the parent process. 
From what I've read this error gets spit out when we try and call the RaspberryPi camera multiple times. 
Maybe this is a race condition?

**Comment by stetsmando** [comment]
So reproducing this is not consistent at all. Suddenly it'll start working. Right now it's working for me and I can't force it to happen again so I've just got to watch and wait. I"m still releasing an update that standardizes the parent/child messaging object so it's easier to do logic on and more consistent. I'll come back to this when I start seeing the error again.

**Comment by stetsmando** [comment]
This should be resolved with v1.1.2

**Comment by stetsmando** [comment]
I lied. It's much less frequent, but it still can happen. I'll continue to look into this.
